cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,544 @@
+import itertools
+from typing import Sequence, Tuple
+from cirq import circuits, devices, ops, protocols, sim, study, value
+import numpy
+from matplotlib import pyplot
+from mpl_toolkits.mplot3d import Axes3D
+
+
+class RabiResult(object):
+    """Class for storing and plotting data from a Rabi oscillation
+    experiment.
+    """
+
+    def __init__(self, rabi_angles: Sequence[float],
+                 excited_state_probs: Sequence[float]):
+        """
+        Args:
+            rabi_angles: The rotation angles of the qubit around the x-axis
+                of the Bloch sphere.
+            excited_state_probs: The corresponding probabilities that the
+                qubit is in the excited state.
+        """
+        self._rabi_angles = rabi_angles
+        self._excited_state_probs = excited_state_probs
+
+    @property
+    def data(self) -> Sequence[Tuple[float, float]]:
+        """Returns a sequence of tuple pairs with the first item being a Rabi
+        angle and the second item being the corresponding excited state
+        probability.
+        """
+        return [(angle, prob) for angle, prob in zip(self._rabi_angles,
+                                                     self._excited_state_probs)]
+
+    def plot(self) -> None:
+        """Plots excited state probability vs the Rabi angle (angle of rotation
+        around the x-axis).
+        """
+        fig = pyplot.figure()
+        pyplot.plot(self._rabi_angles, self._excited_state_probs, 'ro-',
+                    figure=fig)
+        pyplot.xlabel(r"Rabi Angle ($\pi$)")
+        pyplot.ylabel('Excited State Probability')
+
+
+class RBResult(object):
+    """Class for storing and plotting data from a randomized benchmarking
+    experiment.
+    """
+
+    def __init__(self, num_cfds_seq: Sequence[int],
+                 gnd_state_probs: Sequence[float]):
+        """
+        Args:
+            num_cfds_seq: The different numbers of Cliffords in the RB study.
```

Comment:
Don't use abbreviations for API elements exposed to the user. Spell out `ground` and `clifford`.

Reply:
Done